src/goesrecv/config.h

```diff
@@ -44,10 +44,30 @@ struct Config {
     bool bias_tee = 0;
 
     std::unique_ptr<SamplePublisher> samplePublisher;
- };
+  };
 
   Airspy airspy;
 
+  struct HackRf {
+    uint32_t frequency = 0;
+    uint32_t sampleRate = 0;
+
+    // Applies to the tuner IF gain setting
+    uint8_t if_gain = 8;
+
+    // Applies to the tuner BB gain setting
+    uint8_t bb_gain = 30;
+
+    // Enables or disables the RF amplifier
+    bool rf_amp_enabled = 0;
```

Comment:
I hadn't heard of this one. Is it some kind of AGC?

Reply:
Line 62 is for supporting the little LNA package that the HackRF has internally (https://github.com/mossmann/hackrf/blob/master/doc/hardware/hackrf-one-schematic.pdf): https://www.broadcom.com/products/wireless/amplifiers/gain-block-and-drivers/mga-81563

The HackRF wiki has some documentation for the gain settings here: https://github.com/mossmann/hackrf/wiki/FAQ#making-sense-of-gain-settings